lib/plugins/rotate_asg_instances.rb

```diff
@@ -1,16 +1,50 @@
 # frozen_string_literal: true
 
 require 'aws-sdk'
-require_relative 'rotate_asg_instances/asg'
 
 module Moonshot
   module Plugins
     # Rotate ASG instances after update.
     class RotateAsgInstances
+      include ::Moonshot::DoctorHelper
+
+      def doctor(resources)
+        doctor_check_ssh(resources)
+      end
+
+      def pre_update(resources)
+        asg_ssh_class(resources).try_to_ssh
+      end
+
       def post_update(resources)
-        asg = ASG.new(resources)
-        asg.rotate_asg_instances
-        asg.teardown_outdated_instances
+        auto_scaling_group = Moonshot::ASG.new(resources, asg(resources),
+                                               physical_resource_id(resources))
+        auto_scaling_group.rotate_asg_instances
+        auto_scaling_group.teardown_outdated_instances
+      end
+
+      def asg(resources)
+        @asg ||=
+          Aws::AutoScaling::AutoScalingGroup.new(name: physical_resource_id(resources))
+      end
+
+      def asg_ssh_class(resources)
+        @asg_ssh_class ||= Moonshot::Asg::SSH.new(resources)
+      end
+
+      private
+
+      def doctor_check_ssh(resources)
+        asg_ssh_class(resources).try_to_ssh
+        success('Successfully opened SSH connection to an instance.')
+      rescue Moonshot::Asg::SSHValidationError
+        critical('SSH connection test failed, check your SSH settings')
+      end
+
+      def physical_resource_id(resources)
```

Comment:
Why was this method moved out of asg class? I think it belongs there.

Reply:
Moved back